x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/LegacyGeoShapeWithDocValuesIT.java

```diff
@@ -1,109 +0,0 @@
-
-import org.apache.lucene.tests.util.LuceneTestCase;
-import org.elasticsearch.core.UpdateForV9;
-import org.elasticsearch.geometry.Circle;
-import org.elasticsearch.index.IndexVersion;
-import org.elasticsearch.index.IndexVersions;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.search.geo.GeoShapeIntegTestCase;
-import org.elasticsearch.test.index.IndexVersionUtils;
-import org.elasticsearch.xcontent.ToXContent;
-import org.elasticsearch.xcontent.XContentBuilder;
-import org.elasticsearch.xcontent.XContentType;
-import org.elasticsearch.xpack.spatial.LocalStateSpatialPlugin;
-
-import java.io.IOException;
-import java.util.Collection;
-import java.util.Collections;
-
-import static org.elasticsearch.index.query.QueryBuilders.geoShapeQuery;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
-import static org.hamcrest.Matchers.containsString;
-
-@UpdateForV9(owner = UpdateForV9.Owner.SEARCH_ANALYTICS)
-@LuceneTestCase.AwaitsFix(bugUrl = "this is testing legacy functionality so can likely be removed in 9.0")
-public class LegacyGeoShapeWithDocValuesIT extends GeoShapeIntegTestCase {
-
-    @Override
-    protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singleton(LocalStateSpatialPlugin.class);
-    }
-
-    @Override
-    protected void getGeoShapeMapping(XContentBuilder b) throws IOException {
-        b.field("type", "geo_shape");
-        b.field("strategy", "recursive");
-    }
-
-    @Override
-    protected IndexVersion randomSupportedVersion() {
-        // legacy shapes can only be created in version lower than 8.x
-        return IndexVersionUtils.randomPreviousCompatibleVersion(random(), IndexVersions.V_8_0_0);
-    }
-
-    @Override
-    protected boolean allowExpensiveQueries() {
-        return false;
-    }
-
-    public void testMappingUpdate() {
-        // create index
-        assertAcked(
-            indicesAdmin().prepareCreate("test")
-                .setSettings(settings(randomSupportedVersion()).build())
-                .setMapping("shape", "type=geo_shape,strategy=recursive")
-        );
-        ensureGreen();
-
-        String update = """
-            {
-              "properties": {
-                "shape": {
-                  "type": "geo_shape"
-                }
-              }
-            }""";
-
-        IllegalArgumentException e = expectThrows(
-            IllegalArgumentException.class,
-            () -> indicesAdmin().preparePutMapping("test").setSource(update, XContentType.JSON).get()
-        );
-        assertThat(e.getMessage(), containsString("mapper [shape] of type [geo_shape] cannot change strategy from [recursive] to [BKD]"));
-    }
-
-    /**
-     * Test that the circle is still supported for the legacy shapes
-     */
-    public void testLegacyCircle() throws Exception {
```

Comment:
I see this also seems to be tested in [LegacyGeoShapeIT#testLegacyCircle](https://github.com/elastic/elasticsearch/blob/main/modules/legacy-geo/src/internalClusterTest/java/org/elasticsearch/legacygeo/search/LegacyGeoShapeIT.java#L57) so I think we can remove the whole test.

Reply:
Nevermind, will leave this in after talking to @iverase